### ISSUE TITLE
TestCase with DataRow attribute with null parameter ignored

### ISSUE THREAD

**Issue by daangruijters** [microsoft/testfx#1028]
## Description
When using MSTest.TestAdapter version 2.2.4 or higher, DataRow testcases with a null parameter are not discovered or executed. When using 2.2.3, all 3 testcases are discovered.
```
[TestMethod]
[DataRow(null)]
[DataRow("")]
[DataRow(" ")]
public void TestMethod(string parameter)
{

}
```

![image](https://user-images.githubusercontent.com/7239185/147237475-9b0d19b4-26b8-41dc-8d9a-5404c9dfca5d.png)

## Steps to reproduce
Create a test project targetting .NET5 or .NET6, and make sure the version of the package MSTest.TestAdapter is 2.2.4 or higher. Add test method as seen above

## Expected behavior
3 testcases should have been executed, but only the last two are executed

## Actual behavior
Only the testcases with "" and " " as paramter are executed

## Environment
MSTest.TestAdapter 2.2.4 or higher

**Comment by touchofevil2** [comment]
To further the use case this doesn't seem to be anything with null necessarily but the top most attribute seems to be skipped.

It seems the tests are being executed in the reverse order of attributes (bottom one is the first input), and it misses out on the top most DataRow input. eg. In below it will miss out on calling with the null as input.

```
[TestMethod]
[DataRow(null)]
[DataRow("")]
[DataRow(" ")]
public void ValidateString(string? input)
{
    Assert.IsTrue(string.IsNullOrWhiteSpace(input));
    Assert.IsNotNull(signature); // This should have failed for null input
}
```

**Comment by Haplois** [comment]
You need to set `TestDataSourceDiscovery` to `DuringExecution` in your test assembly as described [here](https://github.com/microsoft/testfx/issues/905#issuecomment-907775327).

**Comment by marnilss** [comment]
> You need to set `TestDataSourceDiscovery` to `DuringExecution` in your test assembly as described [here](https://github.com/microsoft/testfx/issues/905#issuecomment-907775327).

Surely that is a temporary workaround, not a solution? (so why close this issue?)

Even though you have changed the implementation of how discovery of tests are done, shouldn't the new one also be fixed to return the expected amount of testcases? Instead of saying - run the old behavior if you want something that works!!!
Because I can't see, nor has anyone else pointed it out, anything wrong with the sample provided by the TS - so it should work even with new discovery method.

**Comment by MikeRosoft** [comment]
Update: apparently the error only occurs when running from Visual Studio; when running from the command line the tests are executed correctly.

**Comment by Evangelink** [comment]
Hi @MikeRosoft, thanks for the update. We will look into this issue.

**Comment by notcake** [comment]
Sounds like a duplicate of #1016, where test cases get ignored when they have the same display name.

**Comment by juharris** [comment]
Thanks @notcake!
Doing
```csharp
[DataRow(null, DisplayName = "null input")]
```
got the test to run in VS for me.

**Comment by Evangelink** [comment]
Fixed through #1306
